smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/httpResponse/HTTPResponseBindingErrorGenerator.kt

```diff
@@ -89,11 +91,20 @@ class HTTPResponseBindingErrorGenerator(
                     writer.write("let data = try await httpResponse.data()")
                     writer.write("let responseReader = try \$N.from(data: data)", ctx.service.readerSymbol)
                     val noErrorWrapping = ctx.service.getTrait<RestXmlTrait>()?.isNoErrorWrapping ?: false
-                    writer.write(
-                        "let baseError = try \$N(httpResponse: httpResponse, responseReader: responseReader, noErrorWrapping: \$L)",
-                        customizations.baseErrorSymbol,
-                        noErrorWrapping
-                    )
+                    if (ctx.service.hasTrait<AwsQueryCompatibleTrait>()) {
+                        writer.write("let errorDetails = httpResponse.headers.value(for: \"x-amzn-query-error\")")
+                        writer.write(
+                            "let baseError = try \$N(httpResponse: httpResponse, responseReader: responseReader, noErrorWrapping: \$L).makeAWSJsonErrorQueryCompatible(errorDetails: errorDetails)",
+                            customizations.baseErrorSymbol,
+                            noErrorWrapping
+                        )
```

Comment:
See my notes in the other PR.  I'd prefer to roll this logic into the `AWSJSONError` constructor.